File: messages-web/src/app/productions/models/OrderByProduct.ts
import { OrderByProduct } from '@/store/catalog-filters.store';

export interface IOrderByProductWithName {
  value: OrderByProduct;
  name: string;
}

export const ordersByProductWithName: IOrderByProductWithName[] = [
  {
    value: OrderByProduct.NameByAsc,
    name: 'по названию (возрастание)',
  },
  {
    value: OrderByProduct.NameByDesc,
    name: 'по названию (убывание)',
  },
  {
    value: OrderByProduct.RegionByAsc,
    name: 'по региону (возрастание)',
  },
  {
    value: OrderByProduct.RegionByDesc,
    name: 'по региону (убывание)',
  },
  {
    value: OrderByProduct.ProducerByAsc,
    name: 'по названию производителя (возрастание)',
  },
  {
    value: OrderByProduct.ProducerByDesc,
    name: 'по названию производителя (убывание)',
  },
  {
    value: OrderByProduct.RatingByAsc,
    name: 'по рейтингу (возрастание)',
  },
  {
    value: OrderByProduct.RatingByDesc,
    name: 'по рейтингу (убывание)',
  },
];

File: Rk.Messages.Spa/Infrastructure/Dto/ProductsNS/OrderByProduct.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Rk.Messages.Spa.Infrastructure.Dto.ProductsNS
{
    /// <summary>
    /// Сортировка продукции по атрибутам
    /// </summary>
    public enum OrderByProduct
    {
        NameByAsc = 0,
        NameByDesc,
        RegionByAsc,
        RegionByDesc,
        ProducerByAsc,
        ProducerByDesc,
        RatingByAsc,
        RatingByDesc,
        IdByAsc,
        IdByDesc
    }
}
